### PR TITLE
MODBULKOPS-334 - Preventing record update with values from different tenants

### PR DESCRIPTION
**Pull request by obozhko-folio**
[MODBULKOPS-334](https://folio-org.atlassian.net/browse/MODBULKOPS-334) - Preventing record update with values from different tenants 

## Purpose
Fix URL relationship

## Approach
Add tenant to relationship id for preview.

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Review by alekGbuz — APPROVED** [review]

File: src/main/java/org/folio/bulkops/service/ElectronicAccessService.java [hunk 1]
```diff
@@ -48,9 +48,9 @@ public ElectronicAccess restoreItemElectronicAccessItem(String access) {
   }
 
   private String electronicAccessToString(ElectronicAccess access, String delimiter) {
-    var tenantId = isNull(access.getTenantId()) ? folioExecutionContext.getTenantId() : access.getTenantId();
+    log.info("electronicAccessToString: {}, {}", access.getRelationshipId(), folioExecutionContext.getTenantId());
```
Comment (khandramai) [inline src/main/java/org/folio/bulkops/service/ElectronicAccessService.java:51]: Remove this logging or make it debug if you really need it.

File: src/main/java/org/folio/bulkops/service/ElectronicAccessService.java [hunk 2]
```diff
@@ -74,4 +74,10 @@ private ElectronicAccess restoreElectronicAccessItem(String access, String delim
     }
     return null;
   }
+
+  private String getRelationshipName(ElectronicAccess access) {
+    log.info("getRelationshipName {}, {}, {}", access.getRelationshipId(), access.getTenantId(), folioExecutionContext.getTenantId());
```
Comment (khandramai) [inline src/main/java/org/folio/bulkops/service/ElectronicAccessService.java:79]: Remove this logging or make it debug if you really need it.

File: src/main/java/org/folio/bulkops/processor/HoldingsDataProcessor.java [hunk 3]
```diff
@@ -210,7 +212,8 @@ private boolean ruleTenantsAreNotValid(BulkOperationRule rule, Action action, Up
     }
     if (nonNull(ruleTenants) && nonNull(actionTenants) && ruleTenants.isEmpty() && actionTenants.isEmpty() &&
       option == ELECTRONIC_ACCESS_URL_RELATIONSHIP && action.getType() == UpdateActionType.FIND_AND_REPLACE) {
-      return true;
+      return !nonNull(extendedHolding.getElectronicAccess()) ||
```
Comment (khandramai) [inline src/main/java/org/folio/bulkops/processor/HoldingsDataProcessor.java:215]: Can this be replaced to isNull?

**Review by khandramai — APPROVED** [review]

**Comment by khandramai** [comment]
LGTM, the only couple of inline comments that will make code more clear.

**Review by siarhei-charniak — APPROVED** [review]

**Comment by obozhko-folio** [comment]
karate_bulkops_succ

File: src/main/java/org/folio/bulkops/processor/HoldingsDataProcessor.java (hunk 3 above)
Comment (obozhko-folio) [reply src/main/java/org/folio/bulkops/processor/HoldingsDataProcessor.java]: Yes, fixed.

File: src/main/java/org/folio/bulkops/service/ElectronicAccessService.java (hunk 2 above)
Comment (obozhko-folio) [reply src/main/java/org/folio/bulkops/service/ElectronicAccessService.java]: Changed to debug.

File: src/main/java/org/folio/bulkops/service/ElectronicAccessService.java (hunk 1 above)
Comment (obozhko-folio) [reply src/main/java/org/folio/bulkops/service/ElectronicAccessService.java]: Changed to debug.

**Pull request merged**